core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLRecordReader.java

```diff
@@ -0,0 +1,305 @@
+
+/**
+ * XMLRecordReader class to read through a given xml document and to output xml blocks as per node path specified.
+ */
+public class XMLRecordReader extends RecordReader<LongWritable, Map<String, String>> {
+  private static final Logger LOG = LoggerFactory.getLogger(XMLRecordReader.class);
+
+  public static final String CLOSING_END_TAG_DELIMITER = ">";
+  public static final String OPENING_END_TAG_DELIMITER = "</";
+  public static final String CLOSING_START_TAG_DELIMITER = ">";
+  public static final String OPENING_START_TAG_DELIMITER = "<";
+
+  private LongWritable currentKey;
+  private Map<String, String> currentValue;
+  private String nodePath;
+  private final String fileName;
+  private final XMLStreamReader reader;
+  private final String[] nodes;
+  private final int totalNodes;
+  private final Map<Integer, String> actualNodeLevelMap;
+  private Map<Integer, String> currentNodeLevelMap;
+  private int nodeLevel = 0;
+  private String tempFilePath = null;
+  private final Path file;
+  private String fileAction;
+  private FileSystem fs;
+  private final String targetFolder;
+  private long availableBytes = 0;
+  private final PublishingInputStream inputStream;
+  private final DecimalFormat df = new DecimalFormat("#.##");
+
+  public XMLRecordReader(FileSplit split, Configuration conf) throws IOException {
+    file = split.getPath();
+    fileName = file.toUri().toString();
+    fs = file.getFileSystem(conf);
+    XMLInputFactory factory = XMLInputFactory.newInstance();
+    FSDataInputStream fdDataInputStream = fs.open(file);
+    inputStream = new PublishingInputStream(fdDataInputStream);
+    availableBytes = inputStream.available();
+    try {
+      reader = factory.createXMLStreamReader(inputStream);
+    } catch (XMLStreamException exception) {
+      throw new RuntimeException("XMLStreamException exception : ", exception);
+    }
+    //set required node path details.
+    nodePath = conf.get(XMLInputFormat.XML_INPUTFORMAT_NODE_PATH);
+    //remove preceding '/' in node path to avoid first unwanted element after split('/')
+    if (nodePath.indexOf("/") == 0) {
+      nodePath = nodePath.substring(1, nodePath.length());
+    }
+    nodes = nodePath.split("/");
+    totalNodes = nodes.length;
+
+    //map to store node path specified nodes with key as node level
+    actualNodeLevelMap = new HashMap<Integer, String>();
+    for (int i = 0; i < totalNodes; i++) {
+      actualNodeLevelMap.put(i, nodes[i]);
+    }
+    currentNodeLevelMap = new HashMap<Integer, String>();
+
+    tempFilePath = conf.get(XMLInputFormat.XML_INPUTFORMAT_PROCESSED_DATA_TEMP_FOLDER);
+    fileAction = conf.get(XMLInputFormat.XML_INPUTFORMAT_FILE_ACTION);
+    targetFolder = conf.get(XMLInputFormat.XML_INPUTFORMAT_TARGET_FOLDER);
+  }
+
+  @Override
+  public void close() throws IOException {
+    if (reader != null) {
+      try {
+        reader.close();
+      } catch (XMLStreamException exception) {
+        // Swallow exception.
```

Comment:
if you want to swallow exception, just use ``Closeables.closeQuietly(reader)``. However, why swallow - perhaps at least log something if there's an exception?